groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/GroovyIndex.java

```diff
@@ -340,14 +332,56 @@ public Set<IndexedMethod> getMethods(final String name, final String clz, QueryS
 
                     // XXX THIS DOES NOT WORK WHEN THERE ARE IDENTICAL SIGNATURES!!!
                     assert map != null;
-                    methods.add(createMethod(signature, map));
+                    IndexedMethod method = createMethod(signature, map);
+                    if (method != null) {
+                        methods.add(method);
+                    }
                 }
             }
         }
 
         return methods;
     }
-
+    
+    // protected for test reasons.
+    protected static boolean matchCamelCase(String prefix, String where, boolean insensitive) {
+        if (where == null || where.length() == 0 || prefix == null || prefix.length() == 0) {
+            return false;
+        }
+        if (!prefix.equals(cachedPrefix) || cachedInsensitive != insensitive) {
```

Comment:
Note the `cachePrefix`, `cachedInsensitive` and `cachedCamelCasePattern` are shared between all instances of `GroovyIndex` across threads (2 threads can call `GroovyIndex.getFields()` simultaneously). Without synchronizing, Java may flush just part of the triplet on write making it temporarily inconsistent for other concurrent accesses.